protocol/tx.go

```diff
@@ -10,17 +10,23 @@ import (
 	"chain/protocol/validation"
 )
 
-// ValidateTxCached checks a cache of prevalidated transactions
-// before attempting to perform a context-free validation of the tx.
-func (c *Chain) ValidateTxCached(tx *bc.Tx) error {
-	// Consult a cache of prevalidated transactions.
-	err, ok := c.prevalidated.lookup(tx.ID)
-	if ok {
+// ErrBadTx is returned for transactions failing validation
+var ErrBadTx = errors.New("invalid transaction")
+
+// ValidateTx validates the given transaction. A cache holds
+// per-transaction validation results and is consulted before
+// performing full validation.
+func (c *Chain) ValidateTx(tx *bc.TxEntries) error {
```

Comment:
Any particular reason for this rename? If the rename is orthogonal to the actual changes of the PR, I'd love it to be in its own PR.